### ISSUE TITLE
In Android For the quip Richtextbox the handleKeyEvent function is not working  

### ISSUE THREAD

**Issue by dharmateja369** [quip/quip-apps#69]
We are Using the handlekeyevent function in richtextbox for the character count this was working fine in ios and browser and desktop quip app. But it was not working on the android so somebody can please help with the issue.

@gabesoft

**Comment by gabesoft** [comment]
@dharmateja369 how would we reproduce the issue? Also, is is broken on all Android versions or a particular one? Also, does it not work when the user types on the keyboard, or when they swipe, or autocorrect? In some of those cases there will be no keyboard event fired, instead input events will be fired.

**Comment by dharmateja369** [comment]
@gabesoft the keyboard event is firing but the problem that I saw when it reached the character limit we are not able to stop the user to type further character limit. So the code was working fine in all other browsers, ios, quip apps for desktop. So to debug can your team can help us on android. You can reproduce this bug was on the v2mom app and the version it was not working on Version 7.8.1 - 708010

**Comment by gabesoft** [comment]
Spend some time debugging this issue with @dharmateja369. Looks like the problem is that on Android calling `preventDefault` on a keyboard event does not actually cancel the event. The event does get fired but it can't be prevented.

**Comment by jesseditson** [comment]
Sorry, finding time to get caught up now. Am I correct in assuming that this is an Android bug rather than a Quip one or did I misread the above comment?

**Comment by gabesoft** [comment]
Yes, it's an Android bug. On Android calling `preventDefault` inside a keydown event with keyCode 229 does not prevents the key from being entered. More info about this issue in this blog post https://clark.engineering/input-on-android-229-unidentified-1d92105b9a04

**Comment by gabesoft** [comment]
But this does looks like a Quip bug on Android because in our [docs](https://quip.com/dev/liveapps/documentation#richtextrecord-actions) we're saying that returning `true` from `handleKeyEvent` prevents the event from happening.

```
handleKeyEvent: PropTypes.func
Callback for keyboard events when this RichTextBox is focused. If handleKeyEvent returns true, the event is not passed to the RichTextBox to be processed
```

**Comment by gabesoft** [comment]
CC @ostacey in case you’re aware of a fix

**Comment by jesseditson** [comment]
This is a pretty in the weeds fix for a small affected audience but the feature detection seems doable based on that blog post. We have a number of accessibility fixes that I’m tempted to group this with, but timeline on those is unclear so far.
